File: mobile/ios/src/data/models/menu.swift
import Foundation

struct Menu: Codable {
  let menuId: Int
  let menuName: String
  let menuDescription: String?
  var products: [Product] = Array()

  init(menuId: Int, menuName: String, menuDescription: String, products list: [Product]?) {
    self.menuId = menuId
    self.menuName = menuName
    self.menuDescription = menuDescription
    self.products = list ?? []
  }
}

File: mobile/android/app/src/main/java/io/meshtech/koasta/net/model/Menu.kt
package io.meshtech.koasta.net.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Menu (
  val menuId: Int,
  val menuName: String,
  val menuDescription: String?,
  val products: List<Product> = emptyList()
)
